ethcore/evm/src/interpreter/gasometer.rs

```diff
@@ -342,6 +347,64 @@ fn add_gas_usize<Gas: evm::CostType>(value: Gas, num: usize) -> (Gas, bool) {
 	value.overflow_add(Gas::from(num))
 }
 
+#[inline]
+fn calculate_eip1283_sstore_gas<Gas: evm::CostType>(schedule: &Schedule, original: &U256, current: &U256, new: &U256) -> Gas {
+	Gas::from(
+		if current == new {
+			schedule.sload_gas
+		} else {
+			if original == current {
+				if original.is_zero() {
+					schedule.sstore_set_gas
+				} else {
+					schedule.sstore_reset_gas
+				}
+			} else {
+				schedule.sload_gas
+			}
+		}
+	)
+}
+
+pub fn handle_eip1283_sstore_clears_refund(ext: &mut vm::Ext, original: &U256, current: &U256, new: &U256) {
```

Comment:
I think there may be issues doing this -- sometimes we need to issue two refunds at once, and the value itself can be either positive or negative. So returning a value instead of directly modifying ext would mean that we need extra codes to handle signed addition/subtraction.